### ISSUE TITLE
JUnit report does not contain failure information

### ISSUE THREAD

**Issue by REDLINK** [Codeception/Codeception#4958]
#### What are you trying to achieve?

- codecept run --xml=result.xml
- Connection to webdriver will terminate for a reason
- see failure in XML

Looks similar to #4675

```xml
<?xml version="1.0" encoding="UTF-8"?>
<testsuites>
  <testsuite name="Tests\Acceptance\Codeception.acceptance" tests="1" assertions="1" errors="0" failures="1" skipped="0" time="1.012901">
    <testcase file="/var/www/test/tests/acceptance/codeception/acceptance/LoginCest.php" name="login" class="Tests\Acceptance\Codeception\Acceptance\LoginCest" feature="login" assertions="1" time="0.698241">
      <failure type="PHPUnit\Framework\ExpectationFailedException">LoginCest: Login
 [ConnectionException] Can't connect to Webdriver at http://127.0.0.1:9515/wd/hub. Please make sure that Selenium Server or PhantomJS is running.
</failure>
    </testcase>
  </testsuite>
</testsuites>
```

#### What do you get instead?

```xml
<?xml version="1.0" encoding="UTF-8"?>
<testsuites>
  <testsuite name="Tests\Acceptance\Codeception.acceptance" tests="1" assertions="0" errors="1" failures="0" skipped="0" time="0.000000">
    <testcase file="/var/www/test/tests/acceptance/codeception/acceptance/LoginCest.php" name="login" class="Tests\Acceptance\Codeception\Acceptance\LoginCest" feature="login" assertions="0" time="0.000000"/>
  </testsuite>
</testsuites>
```
### Details

* Codeception version: 2.3.9
* PHP Version: 7.2.3
* Operating System: Debian 
* Installation type: Composer

**Comment by jared-is-ray** [comment]
I am also able to reproduce this under similar circumstances with Webdriver. If I just do not start selenium which forces the Webdriver connection to fail,  the console output displays the error and even the correct "Errors" count in the summary at the end, but the JUnit xml report fails to include a <failure> element for each test that encounters the error.

As best as I can tell, the following is happening:
- ConnectionException is thrown during the \Codeception\Subscriber\Module::before event phase
- This occurs before \Codeception\PHPUnit\Log\JUnit::startTest is invoked for the test
- The method to actually add the <failure> element is called in (\PHPUnit\Util\Log\JUnit::doAddFault) and fails to create the element because $this->currentTestCase because it should be set by \Codeception\PHPUnit\Log\JUnit::startTest which is never called.

If this is the case, then it seems that this issue might affect any module that throws an exception in the \Codeception\Subscriber\Module::before event phase. In other words, my guess is that this could be hiding multiple types of errors from the JUnit xml reports.

- Codeception version: 2.3.6
- PHP Version: 7.1.17
- OS: Ubuntu Trusty
- Installation type: composer

**Comment by vl-lapikov** [comment]
We also experience this issue, led to false positive reporting.

**Comment by Naktibalda** [comment]
This issue was resolved in Codeception 5.0

I used this command for testing: `./codecept run -c tests/data/exception_in_before --xml`

JUnit report produced by Codeception 4:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<testsuites>
  <testsuite name="unit" tests="4" assertions="0" errors="5" failures="0" skipped="0" time="0.002570">
    <testcase name="failing" file="/.../codeception4/tests/data/exception_in_before/tests/unit/failingCept.php" feature="see that this test was not executed" assertions="0" time="0.000000"/>
    <testcase file="/.../codeception4/tests/data/exception_in_before/tests/unit/failingCest.php" name="failing" class="failingCest" feature="failing" assertions="0" time="0.000000"/>
    <testcase name="testFailing" class="failingTest" file="/.../codeception4/tests/data/exception_in_before/tests/unit/failingTest.php" assertions="0" time="0.002570">
      <error type="RuntimeException">failingTest::testFailing
RuntimeException: in test

/.../codeception4/tests/data/exception_in_before/tests/unit/failingTest.php:14
/.../codeception4/app.php:46
/.../codeception4/app.php:47
/.../codeception4/codecept:7
</error>
    </testcase>
    <testcase file="/.../codeception4/tests/data/exception_in_before/tests/unit/failing.feature" name="failing test: --" feature="failing test" assertions="0" time="0.000000"/>
  </testsuite>
</testsuites>
```

JUnit report produced by Codeception 5:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<testsuites>
  <testsuite name="unit" tests="4" assertions="0" errors="4" failures="0" skipped="0" useless="0" time="0.000000">
    <testcase name="failing" file="/.../Codeception/tests/data/exception_in_before/tests/unit/failingCept.php" feature="see that this test was not executed" time="0.000000" assertions="0">
      <error type="Exception">failingCept: See that this test was not executed
Exception: in before

/.../Codeception/tests/data/exception_in_before/tests/_support/Helper/Unit.php:9
/.../Codeception/vendor/symfony/event-dispatcher/EventDispatcher.php:230
/.../Codeception/vendor/symfony/event-dispatcher/EventDispatcher.php:59
/.../Codeception/app.php:45
/.../Codeception/app.php:46
/.../Codeception/codecept:7
</error>
    </testcase>
    <testcase name="failing" class="FailingCest" file="/.../Codeception/tests/data/exception_in_before/tests/unit/FailingCest.php" feature="failing" time="0.000000" assertions="0">
      <error type="Exception">FailingCest: Failing
Exception: in before

/.../Codeception/tests/data/exception_in_before/tests/_support/Helper/Unit.php:9
/.../Codeception/vendor/symfony/event-dispatcher/EventDispatcher.php:230
/.../Codeception/vendor/symfony/event-dispatcher/EventDispatcher.php:59
/.../Codeception/app.php:45
/.../Codeception/app.php:46
/.../Codeception/codecept:7
</error>
    </testcase>
    <testcase name="testFailing" class="FailingTest" file="/.../Codeception/tests/data/exception_in_before/tests/unit/FailingTest.php" time="0.000000" assertions="0">
      <error type="Exception">FailingTest::testFailing
Exception: in before

/.../Codeception/tests/data/exception_in_before/tests/_support/Helper/Unit.php:9
/.../Codeception/vendor/symfony/event-dispatcher/EventDispatcher.php:230
/.../Codeception/vendor/symfony/event-dispatcher/EventDispatcher.php:59
/.../Codeception/app.php:45
/.../Codeception/app.php:46
/.../Codeception/codecept:7
</error>
    </testcase>
    <testcase name="failing test: --" feature="failing test" file="/.../Codeception/tests/data/exception_in_before/tests/unit/failing.feature" time="0.000000" assertions="0">
      <error type="Exception">failing test: --
Exception: in before

/.../Codeception/tests/data/exception_in_before/tests/_support/Helper/Unit.php:9
/.../Codeception/vendor/symfony/event-dispatcher/EventDispatcher.php:230
/.../Codeception/vendor/symfony/event-dispatcher/EventDispatcher.php:59
/.../Codeception/app.php:45
/.../Codeception/app.php:46
/.../Codeception/codecept:7
</error>
    </testcase>
  </testsuite>
</testsuites>
```
